dynamic_programming/bitmask.py

```diff
@@ -9,77 +9,85 @@
 Find the total no of ways in which the tasks can be distributed.
 """
 from collections import defaultdict
+from typing import List, Dict
 
 
-class AssignmentUsingBitmask:
-    def __init__(self, task_performed, total):
-        self.total_tasks = total  # total no of tasks (N)
+class TaskAssignment:
+    def __init__(self, task_performed: List[List[int]], total_tasks: int) -> None:
+        """
+        Initialize the TaskAssignment class.
 
-        # DP table will have a dimension of (2^M)*N
-        # initially all values are set to -1
+        Parameters:
+        task_performed (List[List[int]]): List of lists where each inner list represents
```

Comment:
Please do not repeat the datatypes that are stated above and checked by mypy.  Readers get confused if one changes and the other does not.
```suggestion
        task_performed: List of lists where each inner list represents
```